### PR TITLE
Add Hamming Distance 

### PR DESCRIPTION
**Pull request by saybarsnazlica**
### **Describe your change:**

- [x] Add an algorithm?
- [ ] Fix a bug or typo in an existing algorithm?
- [ ] Documentation change?

### **Checklist:**

- [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Java/blob/master/CONTRIBUTING.md).
- [x] This pull request is all my own work -- I have not plagiarized.
- [x] I know that pull requests will not be merged if they fail the automated tests.
- [x] This PR only changes one algorithm file. To ease review, please open separate PRs for separate algorithms.
- [x] All new Java files are placed inside an existing directory.
- [x] All filenames are in all uppercase characters with no spaces or dashes.
- [x] All functions and variable names follow Java naming conventions.
- [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.

### CODE REVIEW

File: src/main/java/com/thealgorithms/strings/HammingDistance.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+     * @throws Exception
+     */
+    public static int calculateHammingDistance(String s1, String s2) throws Exception {
+        if (s1.length() != s2.length()) {
+            throw new Exception("String lengths must be equal");
+        }
+
+        int counter = 0;
+        
+        for (int i = 0, length = s1.length(); i < length; i++) {
```
Comment (shreyashHake) [inline src/main/java/com/thealgorithms/strings/HammingDistance.java:24]: Better if `int length = s1.length();` is initialized outside of loop. remaining code looks clean to me .

File: src/main/java/com/thealgorithms/strings/HammingDistance.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    /**
+     * calculate the hamming distance between two strings of equal length
+     *
+     * @param s1 the first string
+     * @param s2 the second string
+     * @return {@code int} hamming distance
+     * @throws Exception
+     */
+    public static int calculateHammingDistance(String s1, String s2) throws Exception {
+        if (s1.length() != s2.length()) {
```
Comment (shreyashHake) [inline src/main/java/com/thealgorithms/strings/HammingDistance.java:18]: Nicely handled.

**Review by shreyashHake — COMMENTED** [review]
Clean code well written..

File: src/main/java/com/thealgorithms/strings/HammingDistance.java (hunk 1 above)
Comment (saybarsnazlica) [reply src/main/java/com/thealgorithms/strings/HammingDistance.java]: Okay, I updated it.

File: src/main/java/com/thealgorithms/strings/HammingDistance.java (hunk 2 above)
Comment (saybarsnazlica) [reply src/main/java/com/thealgorithms/strings/HammingDistance.java:18]: Thank you so much!

**Review by siriak — APPROVED** [review]
Looks good, thanks!

**Comment by saybarsnazlica** [comment]
Thank you!

**Pull request merged**